app/controllers/users.js

```diff
@@ -0,0 +1,31 @@
+const userService = require('../services/users');
+const bcryptService = require('../services/encryption');
+const errors = require('../errors');
+
+const passwordRegex = /^(?=.{8,})(?=.*\d)(?=.*[A-z])(?!.*\s).*$/;
+const emailRegex = /\b[\w\\.-]+@wolox+\.\w{2,4}\b/;
+
+exports.signUp = (req, res, next) => {
+  if (!req.body.firstName) {
```

Comment:
a lot of logic for a controller, move validations to service or a util file, the controller should be clean and simple